Add task list progress header

louwie17 opened the pull request:
Partly addresses #32165 and #32163

Add progress header to task list, this is currently added as a customizable option given it will be used in multiple task lists.

### Screenshots

<img width="798" alt="Screen Shot 2022-03-09 at 4 09 59 PM" src="https://user-images.githubusercontent.com/2240960/157525623-0acc5555-a880-4332-baf3-99ff4455f3d0.png">

### Detailed test instructions:

- Enable the progress header by adding `'display_progress_header' => true` [here](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/OnboardingTasks/TaskLists.php#L93).
- Start a new WooCommerce store and finish the onboarding flow
- Go to **WooCommerce > Home**
- The progress header should show up with a `Welcome to <site name>` 
- The store details will already be set as completed
- Visit a uncompleted task and go back to the home screen, the header should now show 'Let's get you started  🚀'
- Finish three tasks and go back to the home screen
- The header should now say 'You are on the right track' 
- The progress bar should also be updated and the piece of text `4 out of 7 complete` (numbers might be slightly different)
- Now finish 2 more tasks, the header should now say 'You are almost there'
- Now finish the task list, the progress header should disappear.

<!--
Please add your test instructions to `/TESTING-INSTRUCTIONS.md`.
-->

<!--- Please add a Changelog note

Enter a changelog note using the following CLI command `pnpm run changelogger -- add` and commit changes. --->

louwie17 committed:
Add progress header bar to task list experiment

louwie17 committed:
Add changelog

octaedro commented on plugins/woocommerce-admin/client/task-lists/progress-header/progress-header.tsx:
```diff
@@ -0,0 +1,105 @@
+						siteTitle
+				  )
+				: __( 'Welcome', 'woocommerce-admin' );
+		} else if ( completedCount > 0 && completedCount < 3 ) {
+			return __( "Let's get you set up", 'woocommerce-admin' ) + '   🚀';
+		} else if ( completedCount > 2 && completedCount < 5 ) {
```
I think that this `else` and the `else` before are not necessary.

octaedro replied:
Nitpick: the testing instructions said that the header will display the text: `You are on the right track` after finishing the 2nd task, but it will be visible after the 3rd one.

octaedro commented on plugins/woocommerce-admin/client/task-lists/progress-header/progress-header.tsx:
```diff
@@ -0,0 +1,105 @@
+} ) => {
+	const { loading, tasksCount, completedCount, hasVisitedTasks } = useSelect(
+		( select ) => {
+			const isResolving = select( ONBOARDING_STORE_NAME ).isResolving(
```
This is working well but I wonder if using `hasFinishedResolution` instead of `isResolving` would be better here (p90Yrv-1Yq-p2).

octaedro reviewed (commented):
You did a nice job here @louwie17! This is testing well on my end and the code looks mainly good. I just left a couple of small comments.

louwie17 committed:
Address PR feedback and fix some minor issues

louwie17 committed:
Fix task navigation that contains actionUrl

louwie17 commented:
@octaedro thanks for the review :) Are you able to check this out again, I addressed your suggestions in this commit https://github.com/woocommerce/woocommerce/pull/32298/commits/da66d624931450f8fb24740cf072aeb0c1fdc55c (although that commit includes a couple other fixes as well, sorry I should of separated those out).

louwie17 committed:
Fix progress title logic

louwie17 committed:
Remove duplicate property type

octaedro reviewed (approved):
LGTM :shipit:

mattsherman commented:
> Enable the progress header by adding 'display_progress_header' => true [here](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/OnboardingTasks/TaskLists.php#L93).

@woocommerce/mothra It would be best to avoid requiring the tester to modify code in order to test things.
